presto-main/src/main/java/com/facebook/presto/server/ServerInfoResource.java

```diff
@@ -90,11 +92,8 @@ public Response updateState(NodeState state)
                 return Response.ok().build();
             case ACTIVE:
             case INACTIVE:
-                throw new WebApplicationException(Response
-                        .status(BAD_REQUEST)
-                        .type(MediaType.TEXT_PLAIN)
-                        .entity(format("Invalid state transition to %s", state))
-                        .build());
+                nodeState = state;
```

Comment:
What will happen if we request `SHUTTING_DOWN`, then immediately transition to `ACTIVE`/`INACTIVE`?

Reply:
If the node is set to INACTIVE and then SHUTDOWN and we get a new host, the host is ACTIVE and not INACTIVE anymore which is not the desired behavior I guess?

Reply:
> If the node is set to INACTIVE and then SHUTDOWN and we get a new host, the host is ACTIVE and not INACTIVE anymore which is not the desired behavior I guess?

@varungajjala Do you have any concerns about this to support V2 migration?

Reply:
@varungajjala Any other comments? would love to get this merged so that we can integrate with the pipeline.